pandas/core/generic.py

```diff
@@ -3548,7 +3549,7 @@ def to_csv(
         header: bool_t | list[str] = ...,
         index: bool_t = ...,
         index_label: IndexLabel | None = ...,
-        mode: str = ...,
+        mode: Literal["w", "x", "a"] = ...,
```

Comment:
I think many more are allowed for `mode`, `"wt", "wb", ...` not sure whether it is worth narrowing it.

Reply:
I think you are right. One could add all options from build-in ``open``, but maybe it's not worth it.

Character | Meaning
-- | --
'r' | open for reading (default)
'w' | open for writing, truncating the file first
'x' | open for exclusive creation, failing if the file already exists
'a' | open for writing, appending to the end of file if it exists
'b' | binary mode
't' | text mode (default)
'+' | open for updating (reading and writing)

Reply:
Typeshed enumerated all of the combinations. https://github.com/python/typeshed/blob/main/stdlib/_typeshed/__init__.pyi#L139

We might be able to import them inside a TYPE_CHECKING block and then use that in the annotations:

` mode: _typeshed.OpenTextModeUpdating | _typeshed.OpenBinaryModeUpdating = ...,`

(Not sure whether that is a 'good' suggestions)